src/vlq.ml

```diff
@@ -5,81 +5,78 @@
  * LICENSE file in the root directory of this source tree.
  *)
 
-(* VLQ (variable-length quantity) encoder
-   https://en.wikipedia.org/wiki/Variable-length_quantity *)
-
 module type Config = sig
-  val shift: int
-  val char_of_digit: int -> char
-  val digit_of_char: char -> int
+  val shift : int
+  val char_of_int : int -> char
+  val int_of_char : char -> int
 end
 
 module type S = sig
-  val encode: Buffer.t -> int -> unit
-  val decode: char Stream.t -> int
+  val encode : int -> string
+  val decode : string -> int
 end
 
 exception Unexpected_eof
-exception Invalid_base64 of char
+exception Char_of_int_failure of int
+exception Int_of_char_failure of char
 
 module Make (C: Config) = struct
   let vlq_base = 1 lsl C.shift
   let vlq_base_mask = vlq_base - 1
-  let vlq_continuation_bit = vlq_base (* MSB *)
+  let vlq_continuation_bit = vlq_base
 
-  (**
-   * Converts from a two-complement value to a value where the sign bit is
-   * placed in the least significant bit.  For example, as decimals:
-   *   1 becomes 2 (10 binary), -1 becomes 3 (11 binary)
-   *   2 becomes 4 (100 binary), -2 becomes 5 (101 binary)
-   *)
+  (** Converts from a two-complement value to a value where the sign bit is
+      placed in the least significant bit.  For example, as decimals:
+        1 becomes 2 (10 binary), -1 becomes 3 (11 binary)
+        2 becomes 4 (100 binary), -2 becomes 5 (101 binary) *)
   let vlq_signed_of_int value =
-    if value < 0 then ((-value) lsl 1) + 1 else (value lsl 1) + 0
-
-  (* Write the value to the buffer, as multiple characters as necessary *)
-  let rec encode_vlq buf vlq =
-    let digit = vlq land vlq_base_mask in
-    let vlq = vlq lsr C.shift in
-    if vlq = 0 then Buffer.add_char buf (C.char_of_digit digit)
-    else begin
-      (* set the continuation bit *)
-      Buffer.add_char buf (C.char_of_digit (digit lor vlq_continuation_bit));
-      encode_vlq buf vlq
-    end
+      match value < 0 with
+      | true  -> ((-value) lsl 1) + 1
+      | false -> value lsl 1
 
-  (* Encodes `value` as a VLQ and writes it to `buf` *)
-  let encode buf value =
+  let encode value =
     let vlq = vlq_signed_of_int value in
-    encode_vlq buf vlq
+    let rec loop vlq encoded =
+      let digit = vlq land vlq_base_mask in
+      let vlq = vlq lsr C.shift in
+      match vlq = 0 with
+      | true  -> encoded ^ Char.escaped (C.char_of_int digit)
+      | false ->
+        loop vlq (encoded ^ Char.escaped
+          (C.char_of_int (digit lor vlq_continuation_bit))) in
+    loop vlq ""
 
-  let decode =
-    let rec helper (acc, shift) stream =
+  let decode value =
```

Comment:
I see. Thinking in this case, if `encode` receives an `int` and not an `int Stream.t`, then `decode` should work when composed with encode (and vice-versa):

```ocaml
decode (encode x) = x
encode (decode y) = y
```

Mathematically, there is an isomorphism because both functions are bijective, like `string_of_int` and `int_of_string`.

This would be an issue where the sourcemap module would be responsible at all (but, for optimizations, we could expose functions with streams and buffers. I'll implement it.